### PR TITLE
test(augmentation): cover 3D identity matrix behavior

### LINKED ISSUE #4633: AugmentationSequential.identity_matrix is unreachable: 0 calls across 1197 tests
`AugmentationSequential.identity_matrix` (`kornia/augmentation/container/augment.py:395`) overrides the
base implementation to return a `4x4` identity when the container holds 3D augmentations:

```python
def identity_matrix(self, input: torch.Tensor) -> torch.Tensor:
    """Return identity matrix."""
    if self.contains_3d_augmentation:
        return eye_like(4, input)
    return eye_like(3, input)
```

**No call site reaches it.** Every `identity_matrix` call in the package is made on a *child* module, not
on the container. The container-level one is `kornia/augmentation/container/image.py:362`:

```python
mat = module.identity_matrix(input) if _mat is None else _mat
```

— `module`, i.e. the child augmentation, whose own `identity_matrix` comes from
`RigidAffineAugmentationBase2D`/`3D`. The remaining call sites (`_2d/base.py:257`, `:339`,
`_2d/intensity/base.py:131`, `_2d/geometric/pad.py:91`, `_3d/base.py:101`,
`_3d/intensity/equalize.py:77`, `auto/operations/policy.py:96`) are all `self.identity_matrix` inside a
leaf augmentation.

## Measured

Patched the override to count invocations and ran the suite:

```
tests/augmentation/container + tests/augmentation/test_augmentation.py
  --device=cpu --dtype=float32
  1197 passed, 293 skipped, 18 xfailed, 5 xpassed

>>> AugmentationSequential.identity_matrix called 0 times; callers=set()
```

**Zero calls.** That is why a mutation of its predicate (`if self.contains_3d_augmentation` ->
`if not self.contains_3d_augmentation`) survives the whole augmentation suite — I hit this while
mutation-testing #4618, where it was the one mutant of thirteen that lived.

## Why it matters

It is dead code that silently absorbs mutation-testing budget and gives a false impression of coverage
for the 3D container path. Either something *should* be calling it — in which case the 3D container is
returning a `3x3` identity where a `4x4` is intended, and that is a real defect — or it should be
deleted.

Worth deciding which, rather than leaving it. Pre-existing; it is not introduced by any recent PR.

Environment: torch 2.14.0, CPU, macOS arm64, `main` at `3f748920e`.

Posted on behalf of @ducha-aiki by Claude (Opus 5).

### PR DESCRIPTION
**Pull request by NiteeshGowda72**
## Description

Adds regression coverage for the public `AugmentationSequential.identity_matrix()` behavior for 3D augmentations.

`AugmentationSequential` returns a batched 4x4 identity matrix when `contains_3d_augmentation` is true. This test verifies that the existing public behavior remains intact and prevents an accidental regression to the inherited 3x3 implementation.

This PR is intentionally limited to regression test coverage and does not claim to resolve #4633.

## Validation

- Direct 3D identity-matrix regression test: 1 passed
- Full `test_augmentation_sequential.py`: 83 passed, 50 skipped
- Ruff check/format: passed
- Typecheck: no errors
- `git diff --check`: clean

### CODE REVIEW

**Review by ducha-aiki — CHANGES_REQUESTED** [review]
`REQUEST_CHANGES` on `c818de3a3a926928193ed9659afa6b601a2341fe`.

**[P2] Removing the override changes the public 3-D identity from 4×4 to 3×3.** `kornia/augmentation/container/augment.py:391` now lets `AugmentationSequential.identity_matrix` resolve to `ImageSequential.identity_matrix` at `kornia/augmentation/container/image.py:305`, which always returns a 3×3 matrix. The deleted override returned 4×4 whenever `contains_3d_augmentation` was true. This is directly observable even if the current internal transformation path never invokes the method:

```python
import torch
from kornia.augmentation import AugmentationSequential, RandomDepthicalFlip3D

seq = AugmentationSequential(RandomDepthicalFlip3D(p=1.0))
assert seq.contains_3d_augmentation
print(seq.identity_matrix(torch.rand(2, 1, 3, 4, 5)).shape)
```

The reviewed head prints `torch.Size([2, 3, 3])`; the base and the removed implementation return `[2, 4, 4]`. Keep the 3-D dispatch available (or make the inherited implementation dimension-aware) and add a direct regression test before removing this override.

Validation: the focused augmentation-container suite passed on the reviewed head (`794 passed, 233 skipped, 2 xfailed`), which confirms the existing suite does not exercise this public call. The reproduction above imports the reviewed worktree and exposes the behavior change.

Posted on behalf of [@ducha-aiki](https://github.com/ducha-aiki) by Codex (gpt-5.6-sol).

**Comment by NiteeshGowda72** [comment]
Thanks for the review. I’ve addressed the requested changes.

Changes made:

Restored AugmentationSequential.identity_matrix() to preserve the existing public 3D behavior.
Preserved the 4×4 identity matrix when contains_3d_augmentation is True, while retaining the 3×3 identity matrix for 2D augmentations.
Restored the required eye_like import.
Added a direct regression test, test_identity_matrix_3d, that verifies:
3D augmentation detection
returned matrix shape (B, 4, 4)
identity matrix values
input device and dtype preservation

Validation:

Focused regression test: 1 passed
Full test_augmentation_sequential.py: 83 passed, 50 skipped
git diff --check: clean

This also addresses the concern that removing the override changed the publicly observable 3D identity-matrix behavior.

**Review by ducha-aiki — CHANGES_REQUESTED** [review]
REQUEST_CHANGES on `bb9c31b445209a9412442c9d6aa6237a21889915`.

**[P2] Do not publish a removal that the final tree no longer contains — `changelog.d/4640.fixed.md:1`.**

The latest commit restored `AugmentationSequential.identity_matrix` to address the prior public-behavior regression. The net PR diff now contains only this changelog fragment and a regression test; there is no production-code diff, and `kornia/augmentation/container/augment.py:395` still defines the override. Consequently, the fragment, title, description, and `Fixes #4633` claim all describe a change that does not happen.

Either remove the fragment and reframe this as test coverage without closing #4633, or provide another compatible resolution to the issue. As written, merging would publish a false release note and close an unresolved issue.

Validation: the synthetic merge with current `main` (`e428628d0c63e48fd983098bceb647fc799fc433`) is clean; `tests/augmentation/container/test_augmentation_sequential.py --device=cpu --dtype=float32,float64` passed with `169 passed, 100 skipped`. All reported CI checks pass.

Posted on behalf of [@ducha-aiki](https://github.com/ducha-aiki) by Codex (gpt-5.6-sol).

**Comment by NiteeshGowda72** [comment]
## Update

Based on the review feedback, I reverted the production-code removal of `AugmentationSequential.identity_matrix()` because the method is part of the public API and its 3D behavior is observable.

The PR is now intentionally limited to regression test coverage. The added test verifies that:

- `AugmentationSequential` with a 3D augmentation reports `contains_3d_augmentation=True`.
- `identity_matrix()` returns a batched `4×4` identity matrix for 3D inputs.
- The returned matrix preserves the input device and dtype.
- The existing 3D public behavior is protected against an accidental regression to the inherited `3×3` implementation.

### Validation

- Direct 3D regression test: passed
- Full `test_augmentation_sequential.py`: 83 passed, 50 skipped
- Ruff check/format: passed
- Typecheck: no errors
- `git diff --check`: clean

### Changelog check

The `changelog.d/4640.fixed.md` fragment was removed because this PR no longer introduces a user-facing production-code change. The current PR only adds regression test coverage and does not resolve or change the behavior described in #4633.

The `Check changelog fragments` workflow is therefore currently failing because it expects a changelog fragment unless the PR is marked as `no-changelog`.

Since the PR is now test-only, I have intentionally not added a new changelog fragment that would incorrectly describe a user-visible fix. If appropriate, the `no-changelog` label can be applied by a maintainer.

Thanks for the review and guidance.

**Review by ducha-aiki — APPROVED** [review]
Re-reviewed the latest delta. The false fix fragment is removed, and the title/body now accurately describe a test-only regression guard without claiming to resolve #4633. I found no unresolved code findings. The no-changelog label is appropriate; the timed-out float16 job is being rerun before merge.

Posted on behalf of [@ducha-aiki](https://github.com/ducha-aiki) by Codex (gpt-5).
